src/ser.rs

```diff
@@ -10,6 +13,39 @@ use crate::error::{PythonizeError, Result};
 pub trait PythonizeDictType {
     /// Constructor
     fn create_mapping(py: Python) -> PyResult<Bound<PyMapping>>;
+
+    /// Constructor
+    fn create_mapping_with_items<
+        K: ToPyObject,
+        V: ToPyObject,
+        U: ExactSizeIterator<Item = (K, V)>,
+    >(
+        py: Python,
+        items: impl IntoIterator<Item = (K, V), IntoIter = U>,
+    ) -> PyResult<Bound<PyMapping>> {
+        let mapping = Self::create_mapping(py)?;
+
+        for (key, value) in items {
+            mapping.set_item(key, value)?;
+        }
+
+        Ok(mapping)
+    }
+
+    /// Constructor, allows the mappings to be named
+    fn create_mapping_with_items_name<
+        'py,
+        K: ToPyObject,
+        V: ToPyObject,
+        U: ExactSizeIterator<Item = (K, V)>,
+    >(
+        py: Python<'py>,
+        name: &str,
+        items: impl IntoIterator<Item = (K, V), IntoIter = U>,
+    ) -> PyResult<Bound<'py, PyMapping>> {
+        let _name = name;
+        Self::create_mapping_with_items(py, items)
+    }
 }
```

Comment:
I have a couple of thoughts here:
- I wonder, is the split between `create_mapping_with_items` and `create_mapping_with_items_name` a bit confusing? Should we potentially instead model this by having `PythonizeTypes` have a `NamedMap` type to handle the named cases?
- I slightly worry about the performance regression here by adding an intermediate `Vec`. I think the way to solve this would be to replace `create_mapping` with `create_builder(len)`, which returns an associated type implementing `trait MappingBuilder { fn push_item(k, v); fn finish() -> Bound<'py, PyMapping> }`. For the default dict implementation, the builder is then just a thin wrapper around the dict, in the named tuple case the builder can do the pre-collecting into a Vec.

Reply:
I implemented both suggestions. I generally like both, even though they are more infective than the previous change. Perhaps you have some suggestions for how to further improve them?